docs/src/index.md

```diff
@@ -33,7 +34,31 @@ rosenbrock_inp (generic function with 1 method)
 
 ### Reverse mode
 
-The return value of reverse mode is a tuple that contains as a first value
+For scalar-valued functions that take in a single vector, `gradient` and its in-place variant, `gradient!`, provide a simple way to compute only the gradient of `rosenbrock` at a given point:
```

Comment:
I'm hesitant to have this mentioned first as the gradient interface is much more limited than autodiff, and may also come with performance penalties (especially if one makes a closure of a multi arg function instead of calling the multi arg autodiff).

Reply:
Could this be qualified in some way or otherwise strongly encourage the use of autodiff (but have an explanation of the gradient for the sake of adding clarity).

Reply:
I've addressed your feedback by making a separate subsection on "convenience functions" which opens with a disclaimer about flexibility and performance implications. Let me know if you think the performance note is now confusing or overly specific.